src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1673,7 +1673,7 @@ int LinearScan::ComputeOperandDstCount(GenTree* operand)
         // Stores and void-typed operands may be encountered when processing call nodes, which contain
         // pointers to argument setup stores.
         assert(operand->OperIsStore() || operand->OperIsBlkOp() || operand->OperIsPutArgStk() ||
-               operand->OperIsCompare() || operand->OperIs(GT_CMP) || operand->TypeGet() == TYP_VOID);
+               operand->TypeIs(TYP_VOID));
```

Comment:
This `OperIs(GT_CMP)` was seemingly only here because of code in `LowerModPow2` that created a non-void typed `GT_CMP`, but `GT_CMP` does not produce a value so it wasn't expected.
The `OperIsCompare()` check here is unnecessary as those nodes always produce a value now.